src/groupByKey.ts

```diff
@@ -2,6 +2,43 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+interface Student {
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
+}
+
+interface Book {
+  author: string;
+  country: string;
+  imageLink: string;
+  language: string;
+  pages: number;
+  title: string;
+  year: number;
+}
+
+export function groupByKey(
+  items: [],
+  key: string,
+): GroupsMap<Book[] | Student[]> {
```

Comment:
The `items` parameter should have a type that is compatible with both `Book` and `Student` arrays. You can use a generic type with a constraint that ensures the passed `key` is a key of the generic type. This will make the function more type-safe and versatile.